osclib/stagingapi.py

```diff
@@ -807,8 +807,10 @@ def rm_from_prj(self, project, package=None, request_id=None,
 
         for sub_prj, sub_pkg in self.get_sub_packages(package):
             sub_prj = self.map_ring_package_to_subject(project, sub_pkg)
-            if sub_prj != subprj:  # if different to the main package's prj
-                delete_package(self.apiurl, sub_prj, sub_pkg, force=True, msg=msg)
+            # Skip inner-project links for letter stagings as they should
+            # already be in place, but for adi packages remove them.
+            if not self.is_adi_project(project) and sub_prj == project: continue
```

Comment:
I think it should be ``sub_prj == subprj`` due to ``project`` does not go through ``map_ring_package_to_subject()``